### ISSUE TITLE
Grafana Data Source for Elasticsearch type with es_version seems to have incompatible field type

### ISSUE THREAD

**Issue by parker-mar** [grafana/terraform-provider-grafana#243]
### Terraform Version
* Terraform: 1.0.2
* Terraform Grafana Provider: 1.13.1

### Grafana Version
8.0.5

### Affected Resource(s)
* grafana_data_source

### Terraform Configuration Files
```hcl
resource "grafana_data_source" "this" {
  name          = "my-elasticsearch"
  type          = "elasticsearch"
  database_name = "my-database"
  json_data {
    es_version = 70
  }
  ...
}
```

### Debug Output
https://gist.github.com/parker-mar/cdb84c799c87e1b1aee23253fc47c2c2

### Expected Behavior
Resource with es_version created/updated.

### Actual Behavior
Resource with es_version cannot be created/updated (because es_version seems to unmarshal input as string, and passes it along to an int64?).

### Steps to Reproduce
`terraform apply`

### Important Factoids
Under Grafana v8 https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v8-0/#data-source-updates, it seems semver strings are now allowed. I wonder if the Grafana provider is incompatible with this.

> Use semver strings to identify Elasticsearch version
> 
> We changed how the configured Elasticsearch version is handled. You can now specify via provisioning the full semver string version of your instance (such as “7.12.1”) instead of the old version format based on numbers. There’s no manual intervention needed, the old options will be correctly recognized.

For more information, see https://grafana.com/docs/grafana/latest/datasources/elasticsearch/#elasticsearch-version.

**Comment by vtorosyan** [comment]
Thanks for the issue. Indeed with Grafana 8.0+ the change has been introduced which uses semantic versions as string for Elasticsearch. While Grafana backend is compatible with numeric values, this is considered legacy and should not be used anymore, the recommendation is to use full semantic version in the terraform configuration (e.g. `7.0.0` instead of `70`).

[This PR](https://github.com/grafana/terraform-provider-grafana/pull/267) should fix the problem.

**Comment by julienduchesne** [comment]
Should be fixed in v1.14.0+
